pkg/coscheduling/coscheduling.go

```diff
@@ -304,3 +342,59 @@ func (cs *Coscheduling) calculateWaitingPods(podGroupName, namespace string) int
 
 	return waiting
 }
+
+func (cs *Coscheduling) cleanPodGroupInfoIfPresent(obj interface{}) {
+	var pod *v1.Pod
+	switch t := obj.(type) {
+	case *v1.Pod:
+		pod = t
+	case cache.DeletedFinalStateUnknown:
+		var ok bool
+		pod, ok = t.Obj.(*v1.Pod)
+		if !ok {
+			klog.Errorf("cannot convert to *v1.Pod: %v", t.Obj)
+			return
+		}
+	default:
+		klog.Errorf("cannot convert to *v1.Pod: %v", t)
+		return
+	}
```

Comment:
I'm not sure you need to check the error cases since the function is called by PermitPlugin and EventHandler only. The former should pass a valid Pod and the latter already checks the error cases  through FilterFunc.

Reply:
Yes Done